tinyint(1) column type will result in a booleam value in the query data

Project: Sharding-proxy
Version: v4.0.0-RC3

## Question
when the table column type be set to tinyint(1), the query result set of this column will be set to '1'. but it's ok with tinyint(2) or tinyint(3).
This bug seems to be issued in v3.1.0 and i was told that it will be resolved in the later version.

mysql> desc test2;
+-------+------------------+------+------+---------+----------------+
| Field | Type             | Null | Key  | Default | Extra          |
+-------+------------------+------+------+---------+----------------+
| id    | int(10) unsigned | NO   | PRI  | NULL    | auto_increment |
| ctype | **tinyint(1)**       | NO   |      | 0       |                |
+-------+------------------+------+------+---------+----------------+

mysql> insert into test2 values(1,1),(2,2),(3,0),(4,1),(5,2);
Query OK, 5 rows affected (0.13 sec)

In command line mode, no value will return.
mysql> select * from test2;
+------+-------+
| id   | ctype |
+------+-------+
|    1 |      |
|    2 |      |
|    3 |       |
|    4 |      |
|    5 |      |
+------+-------+
5 rows in set (0.03 sec)

In navicate query result, will get a booleam value except value '0'
![微信图片_20191017164947](https://user-images.githubusercontent.com/48242833/66993440-2aaf4600-f0fe-11e9-966f-8684d351aca2.png)

when set to tinyint(2), the query result is ok.
mysql> alter table test2 modify ctype tinyint(2) not null default 0 comment '类型';
Query OK, 0 rows affected (0.06 sec)
mysql> select * from test2;
+------+-------+
| id   | ctype |
+------+-------+
|    1 |     1 |
|    2 |     2 |
|    3 |     0 |
|    4 |     1 |
|    5 |     2 |
+------+-------+
5 rows in set (0.02 sec)
mysql>

Here is the error.

[INFO ] 17:16:42.906 [pool-51-thread-1] ShardingSphere-SQL - SQL: select * from test1 ::: DataSources: slave_okaytest_0
[ERROR] 17:16:42.922 [pool-51-thread-1] o.a.s.s.f.c.CommandExecutorTask - Exception occur: 
java.sql.SQLException: Value '800000-00-00 00:00:00' can not be represented as java.sql.Timestamp
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:964)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:897)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:886)
        at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:860)
        at com.mysql.jdbc.ResultSetRow.getTimestampFast(ResultSetRow.java:937)
        at com.mysql.jdbc.BufferRow.getTimestampFast(BufferRow.java:559)
        at com.mysql.jdbc.ResultSetImpl.getTimestampInternal(ResultSetImpl.java:5918)
        at com.mysql.jdbc.ResultSetImpl.getTimestamp(ResultSetImpl.java:5588)
        at com.zaxxer.hikari.pool.HikariProxyResultSet.getTimestamp(HikariProxyResultSet.java)
        at org.apache.shardingsphere.core.execute.sql.execute.result.QueryResultUtil.getValueByColumnType(QueryResultUtil.java:75)
        at org.apache.shardingsphere.core.execute.sql.execute.result.QueryResultUtil.getValue(QueryResultUtil.java:45)
        at org.apache.shardingsphere.core.execute.sql.execute.result.StreamQueryResult.getValue(StreamQueryResult.java:71)
        at org.apache.shardingsphere.core.merge.dql.common.StreamMergedResult.getValue(StreamMergedResult.java:49)
        at org.apache.shardingsphere.shardingproxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.getQueryData(JDBCDatabaseCommunicationEngine.java:199)
        at org.apache.shardingsphere.shardingproxy.backend.text.query.QueryBackendHandler.getQueryData(QueryBackendHandler.java:64)
        at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.getQueryData(MySQLComQueryPacketExecutor.java:115)
        at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.getQueryData(MySQLComQueryPacketExecutor.java:55)
        at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.MySQLCommandExecuteEngine.writeQueryData(MySQLCommandExecuteEngine.java:83)
        at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:94)
        at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:66)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)

--- a/sharding-proxy/sharding-proxy-backend/src/main/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/datasource/JDBCRawBackendDataSourceFactory.java
+++ b/sharding-proxy/sharding-proxy-backend/src/main/java/org/apache/shardingsphere/shardingproxy/backend/communication/jdbc/datasource/JDBCRawBackendDataSourceFactory.java
@@ -73,6 +73,7 @@ public final class JDBCRawBackendDataSourceFactory implements JDBCBackendDataSou
         config.addDataSourceProperty("elideSetAutoCommits", Boolean.TRUE.toString());
         config.addDataSourceProperty("maintainTimeStats", Boolean.FALSE.toString());
         config.addDataSourceProperty("netTimeoutForStreamingResults", 0);
+        config.addDataSourceProperty("tinyInt1isBit", Boolean.FALSE.toString());
         return new HikariDataSource(config);
     }
     

